cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -211,9 +212,10 @@ def verify_revert_snapshot(full_test_vm, provider, soft_assert, register_event,
     else:
         snapshot1 = new_snapshot(full_test_vm)
     full_template = getattr(provider.data.templates, 'full_template')
+    ssh_ip = wait_pingable(full_test_vm.mgmt, wait=300, allow_ipv6=True)
```

Comment:
You mentioned that this will sometimes get the local IP vs the correct IP, do you want to add a `wait_for` to add that check in?

Reply:
I believe wait_pingable handles that case. It looks for the pingable IP and the 192.x.x.x IP is not pingable AFAIK. This is the behavior I have seen during testing.

Reply:
I have seen the SSH piece pass in several test runs so far.

Reply:
Ah yes, of course! Thanks @prichard77!

Reply:
Waiting for machine being pingable is not good. What should be done is to make ssh trying to repeat the connecting attempt until timeout or until number of attempts exhausted.

The machine can become "pingable" but the service may not be started yet. So this has to be handled in the SSH connection-making code anyway, rendering the `wait_pingable` bit redundant. 

I did some work on the SSH client part about this alredy. I wonder what is the problem. I will take deeper look tomorrow.

Reply:
We do repeat the SSH attempt and handle the exception until timeout. We added the wait_pingable  to wait until we have a pingable IP to populate the SSH arguments. Previously, the hostname in ssh_kwargs was getting populated with None and it would always fail.
I believe the error in PRT we are seeing now may be an intermittent issue I have seen. It didn't show up while testing this locally today. I'll look into it more now.

Reply:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/utility_vm.py#L44
solves a problem of picking the address on which the SSH port is responding and where the `true` command succeeds.

Perhaps this method should be somehow integrated in the ssh client machinery in our code base. There is some logic already and I find it bit intricate IIRC and not sure we have someone brave enough to poke in it at the moment. But we can simply add it and let the rest be.

Note that sometimes I see a "connection reset" on ssh connections when the CFME vm is being configured. I have no idea what is happening. Running SSH connections seem to withstand restarting of SSH daemon, so perhaps some firewall drops the packets.  The linked method of course cannot help with this situation.

Reply:
The code you linked above does look better than what was implemented in the legacy testcase initially. I am considering using it, but if ping will not work then ssh certainly will not work. The test will still timeout trying SSH. There seems to be an underlying issue with the containers or something in PRT as I can run this just fine locally (It fails after SSH for an issue with a BZ opened against it). I am a bit baffled. I wonder if vsphere providers have this issue on PRT? Locally, the wait_pingable was not needed for vsphere providers.

Reply:
@prichard77 Not quite. In the past I found that the `ping` executable is not present in the PRT image. The code executing the ping did bad job reporting what went wrong so it looked like ping didn't pass trough though the real problem was that ping command was not found.

`ping` is only good to test whether the target OS is alive, but that is about all you can infer. It does no guarantee the required service is running, nor whether the service listens on the address you were pinging. It may listen on IPv6 only for example.